src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -88,7 +88,14 @@
       <SourceRoot Include="@(_MappedSourceRoot)" />
     </ItemGroup>
   </Target>
-  
+
+  <!-- 
+    Declare that target InitializeSourceRootMappedPaths that populates MappedPaths metadata on SourceRoot items is available.
+  -->
+  <PropertyGroup>
+    <SourceRootMappedPathsFeatureSupported>true</SourceRootMappedPathsFeatureSupported>
```

Comment:
Do you want to condition on this property not already being set?

Reply:
Why? This should be true if the target is present. It isn't meant to be overridable by the project.

Reply:
Ideally msbuild would have `exists` function that one can pass target name to. This is just a workaround for lack of such feature.

Reply:
https://github.com/Microsoft/msbuild/issues/3090

Reply:
See also https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L748 that uses the same pattern.